x-pack/platform/plugins/shared/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/components/steps/components/package_policy_input_stream.tsx

```diff
@@ -324,6 +330,72 @@ export const PackagePolicyInputStreamConfig = memo<Props>(
+                          <EuiFormRow
+                            label={
+                              <FormattedMessage
+                                id="xpack.fleet.createPackagePolicy.stepConfigure.packagePolicyDataStreamTypeInputLabel"
+                                defaultMessage="Data Stream Type"
+                              />
+                            }
+                            helpText={
+                              isEditPage ? (
+                                <FormattedMessage
+                                  id="xpack.fleet.createPackagePolicy.stepConfigure.packagePolicyInputOnlyEditDataStreamTypeHelpLabel"
+                                  defaultMessage="The data stream type cannot be changed for this integration. Create a new integration policy to use a different input type."
+                                />
+                              ) : (
+                                <FormattedMessage
+                                  id="xpack.fleet.createPackagePolicy.stepConfigure.packagePolicyDataStreamTypeHelpLabel"
+                                  defaultMessage="Select a data stream type for this policy. This setting changes the name of the integration's data stream. {learnMore}."
+                                  values={{
+                                    learnMore: (
+                                      <EuiLink
+                                        href={docLinks.links.fleet.datastreamsNamingScheme}
+                                        target="_blank"
+                                      >
+                                        {i18n.translate(
+                                          'xpack.fleet.createPackagePolicy.stepConfigure.packagePolicyNamespaceHelpLearnMoreLabel',
+                                          { defaultMessage: 'Learn more' }
+                                        )}
+                                      </EuiLink>
+                                    ),
+                                  }}
+                                />
+                              )
+                            }
+                          >
+                            <EuiRadioGroup
+                              data-test-subj="packagePolicyDataStreamType"
+                              disabled={isEditPage}
+                              idSelected={packagePolicyInputStream.data_stream.type || 'logs'}
+                              options={[
+                                {
+                                  id: 'logs',
+                                  label: 'Logs',
+                                },
+                                {
+                                  id: 'metrics',
+                                  label: 'Metrics',
+                                },
+                                {
+                                  id: 'traces',
+                                  label: 'Traces',
+                                },
```

Comment:
Is it ok to allow all types for any input package?

In principle I would say so, it is up to the user to know what kind of data they are collecting.